projects/openapi-utilities/src/utilities/terminal-changelog.ts

```diff
@@ -124,196 +81,209 @@ function* indent(generator: Generator<string>) {
 }
 
 export function* terminalChangelog(
-  groupedChanges: ReturnType<typeof groupChangesAndRules>
+  specs: { from: OpenAPIV3.Document; to: OpenAPIV3.Document },
+  groupedChanges: GroupedDiffs
 ): Generator<string> {
-  const { changesByEndpoint, specification } = groupedChanges;
-  for (const specificationChange of specification.changes) {
-    yield* getDetailLogs(specificationChange, {
-      label: 'specification details:',
-    });
-    yield '';
-  }
-  for (const [_, endpointChange] of changesByEndpoint) {
-    yield* getEndpointLogs(endpointChange);
-  }
-}
-
-function* getEndpointLogs(
-  endpointChange: OpenApiEndpointChange
-): Generator<string> {
-  const {
-    method,
-    path,
-    request,
-    responses,
-    cookieParameters,
-    change,
-    headers,
-    pathParameters,
-    queryParameters,
-  } = endpointChange;
-
-  yield `${chalk.bold(method.toUpperCase())} ${path}: ${
-    change ? getAddedOrRemovedLabel(change) : ''
-  }`;
-
-  if (change?.removed) {
-    return;
-  }
-
-  if (change) {
-    yield* indent(
-      getDetailLogs(change, { excludeKeys: ['pathPattern', 'method'] })
+  const { paths, specification } = groupedChanges;
+  yield* getDetailLogs(specification, {
+    label: 'specification details:',
+  });
+
+  for (const [path, pathObject] of Object.entries(paths)) {
```

Comment:
This felt a little insane to do - i ended up moving the path logic into the groupDiffs function so we don't need to inteprete what "adding a path" means https://github.com/opticdev/optic/pull/1505/commits/6995a8d9939ea7a146dd887629d97717e2a00d59